x-pack/plugins/alerting/server/rule_type_schema.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+const executionStatusErrorReason = schema.oneOf([
+  schema.literal('read'),
+  schema.literal('decrypt'),
+  schema.literal('execute'),
+  schema.literal('unknown'),
+  schema.literal('license'),
+  schema.literal('timeout'),
+  schema.literal('disabled'),
+  schema.literal('validate'),
+]);
+
+const executionStatusWarningReason = schema.oneOf([
+  schema.literal('maxExecutableActions'),
+  schema.literal('maxAlerts'),
+]);
+
+const outcome = schema.oneOf([
+  schema.literal('succeeded'),
+  schema.literal('warning'),
+  schema.literal('failed'),
+]);
+
+export const ruleTypeSchema = schema.object({
```

Comment:
**question:** as the PR evolves, I'm guessing we'll generate `RawRule` TS type definition from this?

Like:
```
import { schema, TypeOf } from '@kbn/config-schema';
...
export type RawRule = TypeOf<typeof ruleTypeSchema>;
```